### DISCUSSION TITLE
Updating gradio - where?

### DISCUSSION THREAD

**Discussion by Sierbahnn** [lllyasviel/Fooocus#2937] [Q&A] [answered]
I got a prompt when I started up stating "IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade."
So I opened up a commandprompt and ran "pip install gradio", and the install ran, placing gradio not in the site-packages drawer but in the python directory.
However, I am still getting the same warning. Am I supposed to install the Gradio while in some other library somewhere? 

Where do I need to install the new gradio for Fooocus to "pick up" on it?

**Comment by ILikeToasters** [comment, marked as answer]
Do not update Gradio. Fooocus needs the version that it is using. Ignore that warning.

**Reply by mashb1t** [reply]
@ILikeToasters thanks!
also see https://github.com/lllyasviel/Fooocus/issues/2911#issuecomment-2108362279

**Reply by Sierbahnn** [reply]
Great. Thank you for your clarification!

**Reply by MohammedSardarSaajit4488** [reply]
thx

**Reply by Mastr-x** [reply]
ty

**Reply by JADYMIR45** [reply]
Muchas graciaspor tu aclaracion buen dia

**Reply by trumpisgood** [reply]
what does Ignore mean if fooocus simply does not run on Google Drive. What to do?
![erorrr foooocus](https://github.com/user-attachments/assets/d7b3ea5c-d40e-4bbe-921d-4d6f85158231)

**Reply by mashb1t** [reply]
@trumpisgood please find the actual RuntimeException in your screenshot above. Gradio 3.X is not the issue here.

**Reply by trumpisgood** [reply]
> please find the actual RuntimeException in your screenshot above. Gradio 3.X is not the issue here.

RuntimeError: Form data requires "python-multipart" to be installed. 
You can install "python-multipart" with: 

pip install python-multipart

model_type EPS
UNet ADM Dimension 2816

But I don't understand what needs to be done?

**Reply by ballboober** [reply]
@trumpisgood use " !pip install --upgrade requests " and then restart de runtime

**Comment by sb2max** [comment]
THX

**Comment by MokkaBoss1** [comment]
I had a go at upgrading Fooocus gradio version to 5.29

https://github.com/MokkaBoss1/Mokkaboss1_Fooocus/blob/main/README.md

**Comment by MohammedSardarSaajit4488** [comment]
do not update the gardio, i caused me problems

**Reply by MokkaBoss1** [reply]
my version works  fine but it took quite a while to reprogram the code

**Reply by MohammedSardarSaajit4488** [reply]
may be it was becos of any other error, idk :)

**Comment by mithunm007** [comment]
which version of Gradio is being currently used in Fooocus

**Comment by MokkaBoss1** [comment]
From memory I think it is 3.34

**Comment by mithunm007** [comment]
what is the required input format for API calls for index 40 41 42, the formats given in the Gradio is raising errors

**Comment by MokkaBoss1** [comment]
Sorry I don’t know the answer to that one
